http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala

```diff
@@ -365,26 +367,33 @@ class GKEInterpreter[F[_]: Parallel: ContextShift: Timer](
       ).compile.lastOrError
 
       // helm install galaxy and wait
+      //TODO: validate app release is the same as retore release
+      galaxyRestore <- persistentDiskQuery.getGalaxyDiskRestore(diskId).transaction
+
       _ <- installGalaxy(
         helmAuthContext,
         app.appName,
         app.release,
+        app.chart,
         dbCluster,
         dbApp.nodepool.nodepoolName,
         namespaceName,
         app.auditInfo.creator,
         app.customEnvironmentVariables,
         ksaName,
-        nfsDisk
+        nfsDisk,
+        galaxyRestore
       )
 
       _ <- logger.info(ctx.loggingCtx)(
         s"Finished app creation for app ${app.appName.value} in cluster ${gkeClusterId.toString}"
       )
 
-      // TODO: this logic will be changed in the next PR
-      isUsedByGalaxy <- persistentDiskQuery.isUsedByGalaxy(diskId).transaction
-      _ <- if (isUsedByGalaxy.exists(identity)) F.unit
+      _ <- if (galaxyRestore.isDefined)
```

Comment:
Even if this is _not_ a restore disk we want to update `lastUsedBy`, right?

Reply:
yea...basically `lastUsedBy` is updated during app creation, if app1 is created using a disk, this disk's `lastUsedBy` becomes this app1

Reply:
Shouldn't we be calling `persistentDiskQuery.updateLastUsedBy()` in the `else` clause below too (since we want to make `lastUpdatedBy = appX` when `appX` is not doing a disk restoration)?

Reply:
it is

Reply:
ah I see